kivy/core/window/__init__.py

```diff
@@ -629,12 +629,30 @@ def _get_android_kheight(self):
             import android
         return android.get_keyboard_height()
 
+    def _get_kivy_vkheight(self):
+        mode = Config.get('kivy', 'keyboard_mode')
+        if (
+            mode in ['dock', 'systemanddock']
```

Comment:
Individual testing would be better as `mode in ['dock', 'systemanddock']` will just create a list to do a contains test and then trash it.

Reply:
I agree with you. Because it's a comparison with only two different values, I don't think there's a big advantage compared to the individual test. Which would be true if there was a test for 5 different values, for example.
I will wait for the opinion of @misl6, before modifying again.

Reply:
At a first glance, could look that `x == "dock" or x =="systemanddock"` is the optimised way to write that condition, but if we analyse that, at least on recents `CPython` implementations, the result is not what could be expected :)

```
>>> from timeit import timeit

>>> x = "systemanddock"
>>> print(timeit(lambda: x == "dock" or x == "systemanddock", number=10000000))
0.8471034320000399
>>> print(timeit(lambda: x in ["dock", "systemanddock"], number=10000000))
0.7669299230000206
>>> print(timeit(lambda: x in ("dock", "systemanddock"), number=10000000))
0.7826992520000431
>>> print(timeit(lambda: x in {"dock", "systemanddock"}, number=10000000))
0.709562496999979

>>> x = "system"
>>> print(timeit(lambda: x == "dock" or x == "systemanddock", number=10000000))
0.8592906410000296
>>> print(timeit(lambda: x in ["dock", "systemanddock"], number=10000000))
0.8776267410000855
>>> print(timeit(lambda: x in ("dock", "systemanddock"), number=10000000))
0.873066920000042
>>> print(timeit(lambda: x in {"dock", "systemanddock"}, number=10000000))
0.7129542200000287 


>>> import dis
>>> dis.dis(compile('x == "dock" or x == "systemanddock"', '<stdin>', 'eval'))
  1           0 LOAD_NAME                0 (x)
              2 LOAD_CONST               0 ('dock')
              4 COMPARE_OP               2 (==)
              6 JUMP_IF_TRUE_OR_POP      8 (to 16)
              8 LOAD_NAME                0 (x)
             10 LOAD_CONST               1 ('systemanddock')
             12 COMPARE_OP               2 (==)
             14 RETURN_VALUE
        >>   16 RETURN_VALUE
>>> dis.dis(compile('x in ["dock", "systemanddock"]', '<stdin>', 'eval'))
  1           0 LOAD_NAME                0 (x)
              2 LOAD_CONST               0 (('dock', 'systemanddock'))
              4 CONTAINS_OP              0
              6 RETURN_VALUE
>>> dis.dis(compile('x in ("dock", "systemanddock")', '<stdin>', 'eval'))
  1           0 LOAD_NAME                0 (x)
              2 LOAD_CONST               0 (('dock', 'systemanddock'))
              4 CONTAINS_OP              0
              6 RETURN_VALUE
>>> dis.dis(compile('x in {"dock", "systemanddock"}', '<stdin>', 'eval'))
  1           0 LOAD_NAME                0 (x)
              2 LOAD_CONST               0 (frozenset({'dock', 'systemanddock'}))
              4 CONTAINS_OP              0
              6 RETURN_VALUE
```

Reply:
Thanks for the clarification @misl6!
So, I think everything is ok with my last commit.

Reply:
@misl6 Nice 👍 

I ran the timeit test on my machine and `x in {"dock, "systemanddock"}` (which compiles to `frozenset`) provides the best performances. 

Python 3.7, Windows 10:
1. Case when function result is true (x = "systemanddock"):
   - 1.782105
   - 1.4844757
   - 1.4534518999999997
   - 1.3753492999999999

2. Case when function result is false (x = "system"):
   - 1.9317027000000002
   - 1.5773074
   - 1.6985533000000004
   - 1.3939842000000002